File: Ywy/Go/level1/최대공약수와최소공배수/gcdAndLcm_test.go
package gcdAndLcm

import (
	"testing"
)

func TestSolution(t *testing.T) {
	arrs := [][]int{
		{3, 12},
		{2, 5},
	}
	except := [][]int{
		{3, 12},
		{1, 10},
	}
	var answer []int
	for index, arr := range arrs {
		answer = Solution(arr[0], arr[1])
		if answer[0] != except[index][0] || answer[1] != except[index][1] {
			t.Errorf("input: %d, answer: %v, except: %v", index, answer, except[index])
		}
	}
}

File: Ywy/JavaScript/level1/test/GcdAndLcm.test.js
import GcdAndLcm from '../GcdAndLcm';

test("1, 3 is [3, 12]", () => {
    expect(new GcdAndLcm().solution(3, 12)).toEqual([3, 12]);
});

test("2, 5 is [1, 10]", () => {
    expect(new GcdAndLcm().solution(2, 5)).toEqual([1, 10]);
});